src/main/scala/sigmastate/serialization/ValueSerializer.scala

```diff
@@ -140,10 +147,16 @@ object ValueSerializer extends SigmaSerializerCompanion[Value[SType]] {
     if (depth > Serializer.MaxTreeDepth)
       throw new ValueDeserializeCallDepthExceeded(s"nested value deserialization call depth($depth) exceeds allowed maximum ${Serializer.MaxTreeDepth}")
     r.level = depth + 1
-    val firstByte = r.peekByte()
-    val v = if (firstByte.toUByte <= LastConstantCode) {
-      // look ahead byte tell us this is going to be a Constant
-      ConstantSerializer(builder).deserialize(r)
+    val firstByte = r.peekByte().toUByte
+    val v = if (firstByte <= LastConstantCode) {
+      if (firstByte == ConstantPlaceholderIndexCode) {
+        // skip opcode
+        r.getByte()
```

Comment:
@greenhat I see Placeholder as very similar to ValUse, the difference is that it takes its values from ConstantsStore, not from the current environment.

Reply:
Done.